### ISSUE TITLE
How to specify a timeout?

### ISSUE THREAD

**Issue by rohanag108** [izumin5210/grapi#101]
I would like to specify a timeout to my RPC calls. Any idea what config needs to be modified and where?

**Comment by izumin5210** [comment]
grapi provides [WithGatewayDialOption](https://godoc.org/github.com/izumin5210/grapi/pkg/grapiserver#WithGatewayDialOptions) to set `grpc.DialOption`.
i think that you can specify a timeout with this option.

**Comment by rohanag108** [comment]
Hey thanks for your response but that did not work:(.
Here is how the server is started:

```golang
s := grapiserver.New(
		grapiserver.WithDefaultLogger(),
                 //grapiserver.WithGatewayDialOptions(grpc.WithTimeout(5*time.Minute)),
		grapiserver.WithGrpcServerOptions(grpc.ConnectionTimeout(5*time.Minute)), //tried this as well
		grapiserver.WithServers(
			server.NewMyServiceServer(),
		),
	)

	return s.Serve()
```
Also seems like `WithTimeout` is deprecated.

I see the following in the logs 
` Failed to write response: write tcp [::1]:3000->[::1]:60638: i/o timeout`

Please let me know if I am missing something. Thanks again for all the help so far.

**Comment by rohanag108** [comment]
@izumin5210 
I found the issue:
https://github.com/izumin5210/grapi/blob/b9d807e0407013ca7b93f624e57c4c14de4fabcf/pkg/grapiserver/gateway.go#L74
![image](https://user-images.githubusercontent.com/4596153/48332206-dbb48300-e607-11e8-828f-1d5e47b25b43.png)

That happens under the covers and there is no lifecycle method to override those values.

**Comment by rohanag108** [comment]
Thanks much appreciated!

**Comment by rohanag108** [comment]
@izumin5210 I see you fixed it v0.3.0 .... thank you so much!. Just waiting for you to release the version!

**Comment by izumin5210** [comment]
v0.3.0 released 👍
